src/Stratis.Bitcoin.Tests/Utilities/ParallelAsyncTest.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Utilities;
+using Xunit;
+
+namespace Stratis.Bitcoin.Tests.Utilities
+{
+    public class ParallelAsyncTest
```

Comment:
Can we maybe add test that checks the newly added cancellation token?

Reply:
Oh yeah, sure.

Reply:
Fixed.